cpp/src/Slice/Scanner.l

```diff
@@ -693,6 +696,36 @@ namespace
         }
     }
 
+    void printStateStack()
+    {
+        fprintf(stderr, "current state stack [ " );
+        for (int i = 0; i < yy_start_stack_ptr; i++)
+        {
+            fprintf(stderr, "%d ", yy_start_stack[i]);
+        }
+        fprintf(stderr, "(%d) ]\n", YY_START);
+    }
+
+    void pushState(int newState)
+    {
+        yy_push_state(newState);
+        if (yy_flex_debug)
```

Comment:
This `yy_flex_debug` bool is controlled by passing `--debug` on the command line.
Ex: `slice2cpp --debug Test.ice`.